x-pack/plugins/siem/public/components/open_timeline/types.ts

```diff
@@ -0,0 +1,161 @@
+export type OnToggleShowNotes = (itemIdToExpandedNotesRowMap: Record<string, JSX.Element>) => void;
+
+/** Parameters to the OnTableChange callback  */
+export interface OnTableChangeParams {
+  page: {
+    index: number;
+    size: number;
+  };
+  sort: {
+    field: string;
+    direction: 'asc' | 'desc';
+  };
+}
+
+/** Invoked by the EUI table implementation when the user interacts with the table */
+export type OnTableChange = (tableChange: OnTableChangeParams) => void;
+
+export interface OpenTimelineProps {
+  /** Invoked when the user clicks the delete (trash) icon on an individual timeline */
+  deleteTimelines?: DeleteTimelines;
+  /** The default requested size of each page of search results */
+  defaultPageSize: number;
+  /** Displays an indicator that data is loading when true */
+  isLoading: boolean;
+  /** Required by EuiTable for expandable rows: a map of `TimelineResult.savedObjectId` to rendered notes */
+  itemIdToExpandedNotesRowMap: Record<string, JSX.Element>;
+  /** If this callback is specified, a "Favorite Selected" button will be displayed, and this callback will be invoked when the button is clicked */
+  onAddTimelinesToFavorites?: OnAddTimelinesToFavorites;
+  /** If this callback is specified, a "Delete Selected" button will be displayed, and this callback will be invoked when the button is clicked */
+  onDeleteSelected?: OnDeleteSelected;
+  /** Only show favorite timelines when true */
+  onlyFavorites: boolean;
+  /** Invoked when the user presses enter after typing in the search bar */
+  onQueryChange: OnQueryChange;
+  /** Invoked when the user selects (or de-selects) timelines in the table */
+  onSelectionChange: OnSelectionChange;
+  /** Invoked when the user clicks on the name of a timeline to open it */
+  onOpenTimeline: OnOpenTimeline;
+  /** Invoked by the EUI table implementation when the user interacts with the table */
+  onTableChange: OnTableChange;
+  /** Invoked when the user toggles the option to only show favorite timelines */
+  onToggleOnlyFavorites: OnToggleOnlyFavorites;
+  /** Invoked when the user toggles the expansion or collapse of inline notes in a table row */
+  onToggleShowNotes: OnToggleShowNotes;
+  /** the requested page of results */
+  pageIndex: number;
+  /** the requested size of each page of search results */
+  pageSize: number;
+  /** The currently applied search criteria */
+  query: string;
+  /** The results of executing a search */
+  searchResults: TimelineResult[];
+  /** the currently-selected timelines in the table */
+  selectedItems: TimelineResult[];
+  /** the requested sort direction of the query results */
+  sortDirection: 'asc' | 'desc';
+  /** the requested field to sort on */
+  sortField: string;
+  /** The title of the Open Timeline component  */
+  title: string;
+  /** The total (server-side) count of the search results */
+  totalSearchResultsCount: number;
+}
+
+/** A type that implements `isDisabled` (used for testing) */
```

Comment:
Per my comments above, I would recommend removing these in favor of using the actual types:
https://github.com/elastic/kibana/pull/35782/files#r280194126